samples/fsharp/getting-started/AdvancedExperiment_AutoML/AdvancedTaxiFarePrediction/Program.fs

```diff
@@ -0,0 +1,336 @@
+    let pinfo = new ProcessStartInfo(chartFileNamePath, UseShellExecute = true)
+    Process.Start pinfo |> ignore
+
+
+
+let mlContext = new MLContext()
+
+// Infer columns in the dataset with AutoML
+let columnInference = 
+    ConsoleHelper.consoleWriteHeader "=============== Inferring columns in dataset ==============="
+    let columnInference = mlContext.Auto().InferColumns(trainDataPath, labelColumnName, groupColumns=false)
+    ConsoleHelper.print columnInference
+    columnInference
+
+// Load data from files using inferred columns
+let textLoader = mlContext.Data.CreateTextLoader(columnInference.TextLoaderOptions)
+let trainDataView = textLoader.Load(trainDataPath)
+let testDataView = textLoader.Load(testDataPath)
+
+// Run an AutoML experiment on the dataset
+let experimentResult = 
+
+    // STEP 1: Display first few rows of the training data 
+    ConsoleHelper.showDataViewInConsole mlContext trainDataView 4
+
+    // STEP 2: Build a pre-featurizer for use in the AutoML experiment.
+    // (Internally, AutoML uses one or more train/validation data splits to 
+    // evaluate the models it produces. The pre-featurizer is fit only on the 
+    // training data split to produce a trained transform. Then, the trained transform 
+    // is applied to both the train and validation data splits.)
+    let preFeaturizer = mlContext.Transforms.Conversion.MapValue("is_cash", [| KeyValuePair("CSH", true) |], "payment_type") |> downcastPipeline
+
+    // STEP 3: Customize column information returned by InferColumns API
+    let columnInformation = columnInference.ColumnInformation
+    columnInformation.CategoricalColumnNames.Remove("payment_type") |> ignore
+    columnInformation.IgnoredColumnNames.Add("payment_type")
+
+    // STEP 4: Initialize a cancellation token source to stop the experiment.
+    let cts = new CancellationTokenSource()
+
+    // STEP 5: Initialize our user-defined progress handler that AutoML will 
+    // invoke after each model it produces and evaluates.
+    let progressHandler = ConsoleHelper.regressionExperimentProgressHandler()
+
+    // STEP 6: Create experiment settings
+    let experimentSettings = 
+        let experimentSettings = new RegressionExperimentSettings()
+        experimentSettings.MaxExperimentTimeInSeconds <- 3600u
+        experimentSettings.CancellationToken <- cts.Token
+
+        // Set the metric that AutoML will try to optimize over the course of the experiment.
+        experimentSettings.OptimizingMetric <- RegressionMetric.RootMeanSquaredError
+
+        // Set the cache directory to null.
+        // This will cause all models produced by AutoML to be kept in memory 
+        // instead of written to disk after each run, as AutoML is training.
+        // (Please note: for an experiment on a large dataset, opting to keep all 
+        // models trained by AutoML in memory could cause your system to run out 
+        // of memory.)
+        experimentSettings.CacheDirectory <- null
```

Comment:
Setting to `null` is generally a bad idea. Not sure why in the C# version we first rather warn to not set to null, and then set to null. Mainly this limits the number of iterations a user can run.

Reply:
```suggestion
        //experimentSettings.CacheDirectory <- null
```